src/librustc_codegen_ssa/back/linker.rs

```diff
@@ -28,7 +28,9 @@ use rustc_target::spec::{LinkOutputKind, LinkerFlavor, LldFlavor};
 pub fn disable_localization(linker: &mut Command) {
     // No harm in setting both env vars simultaneously.
     // Unix-style linkers.
-    linker.env("LC_ALL", "C");
+    // We use an UTF-8 locale, as the generic C locale disables support for non-ASCII
+    // bytes in filenames on some platforms.
+    linker.env("LC_ALL", "en_US.UTF-8");
```

Comment:
Wouldn't `C.UTF-8` be better here?

Reply:
`C.UTF-8` is a non-standard locale invented by Debian: https://github.com/commercialhaskell/stack/issues/856

Reply:
C.UTF-8 is making its way upstream.